### PR TITLE
Basic dialog

### PR DESCRIPTION
**Pull request by rogerkuou**
Created an initial dialog, which takes 1 raster layer and 2 vector layers in QGIS, converting raser layer to an `Xarray.Dataset`, and the vector layers `GeoDataFrame`.

### CODE REVIEW

File: droneml_dialog.py [hunk 1]
```diff
@@ -1,81 +1,144 @@
...
+        )[0]
+        vec_negative_layer = QgsProject.instance().mapLayersByName(
+            self.vec_negative_combo.currentText()
+        )[0]
+
+        print(f"Raster Layer: {raster_layer}")
+        print(f"Positive Vector Layer: {vec_positive_layer}")
+        print(f"Negative Vector Layer: {vec_negative_layer}")
+
+        ds_raster = rioxarray.open_rasterio(raster_layer.source())
```
Comment (cwmeijer) [inline droneml_dialog.py:91]: non tif-file data source (such as some tiled data providindg service)

File: droneml_dialog.py [hunk 2]
```diff
@@ -1,81 +1,144 @@
...
-        self.canvas.setCanvasColor(Qt.white)
-        map_settings = QgsMapSettings()
-        self.canvas.setExtent(map_settings.extent())
-        layout.addWidget(self.canvas)
+        # Create a layout to organize widgets in the dialog
+        self.layout = QtWidgets.QVBoxLayout()
+
+        # Combo box for raster layers
+        # Label
+        self.raster_label = QtWidgets.QLabel("Raster layer for training:")
```
Comment (cwmeijer) [inline droneml_dialog.py:27]: code duplication (3x label+combobox)

**Review by cwmeijer — APPROVED** [review]

File: droneml_dialog.py (hunk 2 above)
Comment (rogerkuou) [reply droneml_dialog.py:27]: doced this in issue #6

File: droneml_dialog.py (hunk 1 above)
Comment (rogerkuou) [reply droneml_dialog.py:91]: doced this in issue #7

**Pull request merged**